zerocopy-derive/src/enum.rs

```diff
@@ -274,6 +282,63 @@ pub(crate) fn derive_is_bit_valid(
             )
             .inner_extras(quote! {
                 type Type = #ty;
+
+                #[inline(always)]
+                fn project(slf: #zerocopy_crate::PtrInner<'_, Self>) -> #zerocopy_crate::PtrInner<'_, Self::Type> {
```

Comment:
Consider replacing the logic here with a use of the macro introduced in #2854

Reply:
Resolved by introducing `PtrInner::project`